### PR TITLE
refactor: set target pressures in compressor train model and compressor stages for failure status evaluations

### PR DESCRIPTION
**Pull request by olelod**
## What does this pull request change?

- Added target_suction_pressure, target_discharge_pressure to CompressorTrainStage and also and target_intermediate_pressure to CompressorTrainModel.
- Added target_pressure_status to CompressorTrainStageResultSingleTimeStep. This will show if the individual stages are able to meet the defined target pressures (Most often this will be the target_suction_pressure of the first compressor stage and the target_discharge_pressure of the last compressor stage. The other pressures are free to vary). target_pressure_status can be: ABOVE/BEWOW_TARGET_SUCTION/DISCHARGE_PRESSURE, NOT_CALCULATED or TARGET_PRESSURES_MET. 
- Added within_capacity to CompressorTrainResultSingleTimeStep and CompressorTrainStageResultSingleTimeStep. This will show if the compressor stage operates within capacity (inside the chart, or inside the chart after recirculation). This was formerly what was used as is_valid for a compressor stage.
- Is_valid for CompressorTrainStageResultSingleTimeStep is now the combination of target_pressure_status and within_capacity. The compressor stage is only valid if it satisfies the target pressures AND operates within capacity.
- If one stage in a compressor train is not valid, the compressor train model will be not valid. The failure_status property on the CompressorTrainResultSingleTimeStep will tell what the problem is.

For single speed compressor trains, the looping over time steps have been moved to the evaluate_rate_ps_pd function, like it is for most other compressor train models.

This should also allow for re-introducing is_valid plots for individual compressor stages (ECALC-827)

### CODE REVIEW

File: src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json [hunk 1]
```diff
@@ -19983,9 +19983,9 @@
                             ],
                             "power_unit": "MW",
                             "pressure_is_choked": [
-                                false,
-                                false,
-                                false,
+                                true,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json:19986]: This is upstream choking. Should be true. Was not before. Fixed now.

File: src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json [hunk 2]
```diff
@@ -18097,9 +18097,9 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json:18100]: This is upstream choking. Should be true. Was not before. Fixed now.

File: src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json [hunk 3]
```diff
@@ -17428,9 +17428,9 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json:17431]: This is upstream choking. Should be true. Was not before. Fixed now.

File: src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json [hunk 4]
```diff
@@ -17003,9 +17003,9 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json:17006]: This is upstream choking. Should be true. Was not before. Fixed now.

File: src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json [hunk 5]
```diff
@@ -17310,9 +17310,9 @@
                                     NaN
                                 ],
                                 "pressure_before_choking": [
-                                    75.80978,
-                                    75.80978,
-                                    75.80978,
+                                    80.0,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_energy_usage_models/test_all_results/all_energy_usage_models_v3.json:17313]: Pressure_before_choking was wrong before. Should be 80 here. Fixed now.

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 6]
```diff
@@ -630,8 +630,8 @@
                                 ]
                             },
                             "is_valid": [
-                                true,
-                                true
+                                false,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:633]: Here the stage results appear as is_valid == False when the stage does not meet the requested pressure. The stage used to be valid if it was within capacity. Now that has changed.

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 7]
```diff
@@ -2792,12 +2792,12 @@
                                 ]
                             },
                             "is_valid": [
-                                true,
-                                true,
-                                true,
-                                true,
-                                true,
-                                true
+                                false,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:2795]: Here the stage results appear as is_valid == False when the stage does not meet the requested pressure. The stage used to be valid if it was within capacity. Now that has changed.

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 8]
```diff
@@ -7116,17 +7116,17 @@
...
-                                        false,
-                                        false,
-                                        false,
-                                        false,
-                                        false,
-                                        false,
-                                        false,
-                                        false,
-                                        false
+                                        true,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:7119]: downstream choking now correctly reported with pressure_is_choked == True

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 9]
```diff
@@ -10057,15 +10057,15 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:10060]: downstream choking now correctly reported with pressure_is_choked == True

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 10]
```diff
@@ -11122,17 +11122,17 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:11125]: downstream choking now correctly reported with pressure_is_choked == True

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 11]
```diff
@@ -12197,17 +12197,17 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:12200]: downstream choking now correctly reported with pressure_is_choked == True

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 12]
```diff
@@ -13255,15 +13255,15 @@
...
-                                false,
-                                false,
-                                false,
-                                false,
-                                false,
-                                false,
-                                false,
-                                false,
-                                false
+                                true,
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:13258]: downstream choking now correctly reported with pressure_is_choked == True

File: src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json [hunk 13]
```diff
@@ -15247,17 +15247,17 @@
...
```
Comment (olelod) [inline src/tests/libecalc/integration/snapshots/test_all_consumer_with_time_slots_models/test_all_consumer_with_time_slots_models_results/all_consumer_with_time_slots_models_v3.json:15250]: downstream choking now correctly reported with pressure_is_choked == True

**Review by jsolaas — APPROVED** [review]

**Comment by TeeeJay** [comment]
nice work :)

**Pull request merged**